src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1871,12 +1860,23 @@ private BetterResult BetterConversionFromExpression(BoundExpression node, TypeSy
                         return conv2.Kind == ConversionKind.Identity ? BetterResult.Right : BetterResult.Neither;
                     }
                 }
-                else if (refKind1 == RefKind.Ref)
+                else if (p1.RefKind == RefKind.Ref)
                 {
                     return BetterResult.Neither;
                 }
             }
 
+            // If an ambiguity exists between 'by-val' and 'in' parameter, choose the 'by-val' one, except if it was params,
```

Comment:
>// If an ambiguity exists between 'by-val' and 'in' parameter, choose the 'by-val' one, except if it was params, [](start = 12, length = 112)

It feels suspicious that we do this check before we checked which candidate uses BetterConversionFromExpression. Even if we do that immediately after, we might still introduce a breaking change as the tie breaking rules that are executed today after this function can resolve the ambiguity differently.